lib/storage/src/rbac/mod.rs

```diff
@@ -1,9 +1,209 @@
-pub mod access;
-pub mod access_level;
-pub mod access_token;
-#[allow(dead_code)]
-pub mod collection_access;
-#[allow(dead_code)]
-pub mod error;
-
 mod ops_checks;
+
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+use segment::json_path::JsonPath;
+use segment::types::ValueVariants;
+use serde::{Deserialize, Serialize};
+
+use crate::content_manager::errors::StorageError;
+
+/// A structure that defines access rights.
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
+#[serde(untagged)]
+pub enum Access {
+    /// Cluster-wide access.
+    Cluster(ClusterAccessMode),
+    /// Access to specific collections.
+    Collection(AccessCollection),
```

Comment:
Other renaming suggestions: 
`AccessCollection` -> `CollectionAccesses`
`AccessCollectionRule` -> `CollectionAccess`

and to avoid repeating `access` in claims:
`AccessCollectionRule.access` -> `AccessCollectionRule.mode`?

Reply:
Renamed to `CollectionAccess`/`CollectionAccessList`/`CollectionAccessView`.

I'd keep `AccessCollectionRule.access` because that's how it's named in JSON.